simapp/encoding.go

```diff
@@ -1,16 +1,12 @@
 package simapp
 
 import (
-	"github.com/cosmos/cosmos-sdk/simapp/params"
-	"github.com/cosmos/cosmos-sdk/std"
+	simapparams "github.com/cosmos/cosmos-sdk/simapp/params"
 )
 
-// MakeEncodingConfig creates an EncodingConfig for testing
-func MakeEncodingConfig() params.EncodingConfig {
-	encodingConfig := params.MakeEncodingConfig()
-	std.RegisterLegacyAminoCodec(encodingConfig.Amino)
-	std.RegisterInterfaces(encodingConfig.InterfaceRegistry)
-	ModuleBasics.RegisterLegacyAminoCodec(encodingConfig.Amino)
-	ModuleBasics.RegisterInterfaces(encodingConfig.InterfaceRegistry)
+// MakeEncodingConfigTests creates an EncodingConfig for testing
+func MakeEncodingConfigTests() simapparams.EncodingConfig {
```

Comment:
I believe most (new) app developers take `simapp` as a reference (aka copy/paste, and then tweak some modules). Calling this Tests can be quite confusing.

Reply:
I added a `Test` suffix to differentiate it from the `params.MakeProtoEncodingConfig`. I was inspired by the comment of that function ("... for testing"). I'm happy to hear alternative proposals.

Reply:
I think it's really more accurately "for testing and simulations"

I think calling it something like `MakeEncodingConfigWithInterfaceRegistry` or `MakeEncodingConfigWithRegistry` is a bit more clear, even though its long.

Reply:
I would propose to rename back to `MakeEncodingConfig` to make this PR slightly more digestable. We should definitely revisit this naming (and imo also in which folder it lives).

Reply:
The function already has changes and is creating problems in other places. I would still prefer to rename it and have it in Stargate release, instead of starting a new discussion and porting it back to Stargate. 

If the function is indeed used in "production" then how about this name: `MakeAndRegEncodingConfig`